internal/app/machined/pkg/system/services/etcd.go

```diff
@@ -439,7 +424,142 @@ func (e *Etcd) args(r runtime.Runtime) ([]string, error) {
 		blackListArgs.Set("advertise-client-urls", fmt.Sprintf("https://%s:2379", ips[0].String()))
 	}
 
-	return blackListArgs.Merge(extraArgs).Args(), nil
+	e.args = blackListArgs.Merge(extraArgs).Args()
+
+	return nil
+}
+
+func (e *Etcd) setup(ctx context.Context, r runtime.Runtime, errCh chan error) {
+	var err error
+
+	if err = os.MkdirAll(constants.EtcdDataPath, 0755); err != nil {
+		errCh <- err
+
+		return
+	}
+
+	if err = generatePKI(r); err != nil {
+		errCh <- fmt.Errorf("failed to generate etcd PKI: %w", err)
+
+		return
+	}
+
+	client, err := containerdapi.New(constants.ContainerdAddress)
+	if err != nil {
+		errCh <- err
+
+		return
+	}
+	// nolint: errcheck
+	defer client.Close()
+
+	// Pull the image and unpack it.
+	containerdctx := namespaces.WithNamespace(ctx, constants.SystemContainerdNamespace)
+	if _, err = image.Pull(containerdctx, r.Config().Machine().Registries(), client, r.Config().Cluster().Etcd().Image()); err != nil {
+		errCh <- fmt.Errorf("failed to pull image %q: %w", r.Config().Cluster().Etcd().Image(), err)
+	}
+
+	switch r.Config().Machine().Type() {
+	case runtime.MachineTypeInit:
+		err = e.argsForInit(r)
+		if err != nil {
+			errCh <- err
+		}
+	case runtime.MachineTypeControlPlane:
+		err = e.argsForControlPlane(r)
+		if err != nil {
+			errCh <- err
+		}
+	}
```

Comment:
We need this still because we have to handle the init node differently. Goal is to get rid of the init node, but we have to keep it around for some time.